Doc/library/io.rst

```diff
@@ -911,21 +923,22 @@ Text I/O
 
    .. method:: seek(offset, whence=SEEK_SET, /)
 
-      Change the stream position to the given *offset*.  Behaviour depends on
-      the *whence* parameter.  The default value for *whence* is
-      :data:`SEEK_SET`.
-
-      * :data:`SEEK_SET` or ``0``: seek from the start of the stream
-        (the default); *offset* must either be a number returned by
-        :meth:`TextIOBase.tell`, or zero.  Any other *offset* value
-        produces undefined behaviour.
-      * :data:`SEEK_CUR` or ``1``: "seek" to the current position;
-        *offset* must be zero, which is a no-operation (all other values
-        are unsupported).
-      * :data:`SEEK_END` or ``2``: seek to the end of the stream;
-        *offset* must be zero (all other values are unsupported).
-
-      Return the new absolute position as an opaque number.
+      Change the stream position to the given byte *offset*,
+      and return the new absolute position as an opaque number.
+      *offset* is interpreted relative to the position indicated by *whence*.
+      Valid values for *whence* are:
+
+      * :data:`SEEK_SET` -- seek from the start of the stream (the default);
+        *offset* must either be a number returned by
+        :meth:`TextIOBase.tell`, or zero.
+        Any other *offset* value produces undefined behaviour.
+      * :data:`SEEK_CUR` -- seek from current stream position;
+        *offset* must be zero, which is a no-operation.
```

Comment:
```suggestion
      * :data:`SEEK_CUR` -- just return the current position; *offset* must be zero.
```

Reply:
Why?

Reply:
Isn't it silly? "Seek, but offset must be zero, so actually do not seek." Why write seek if it does not seek? I think it only confuses users.

Reply:
I suggested "seek from the current stream position; *offset* must be zero, which does not change the stream position. All other values are unsupported, and will raise an exception." above -- would this be an alright alternative?

Reply:
Why say "seek" if it does not seek?

Reply:
Up to you. Only do not include contradictions.

Since `seek()` also returns the current position in all other cases, "return the current position" may be superfluous (it stress the difference which does not exist). Maybe just "no-operation", "does nothing" or "does not change the current position"?

Reply:
> Maybe just "no-operation", "does nothing" or "does not change the current position"?

Yes, I think a shorter sentence is the way to go.